src/commands/add_command/access_key/sender/mod.rs

```diff
@@ -47,12 +59,28 @@ impl Sender {
 }
 
 impl Sender {
-    pub fn input_sender_account_id() -> String {
-        println!();
-        Input::new()
-            .with_prompt("What is the account ID of the sender?")
-            .interact_text()
-            .unwrap()
+    fn input_sender_account_id(
+        connection_config: Option<crate::common::ConnectionConfig>,
+    ) -> color_eyre::eyre::Result<String> {
+        match &connection_config {
+            Some(connection_config) => loop {
+                let account_id: String = Input::new()
+                    .with_prompt("What account ID do you need to add a key?")
+                    .interact_text()
+                    .unwrap();
+                match crate::common::check_account_id(
+                    connection_config.clone(),
+                    account_id.clone(),
+                )? {
+                    Some(_) => break Ok(account_id),
+                    None => println!("This account ID <{}> doesn't exist", account_id),
+                };
+            },
+            None => Ok(Input::new()
+                .with_prompt("What account ID do you need to add a key?")
+                .interact_text()
+                .unwrap()),
+        }
```

Comment:
It is less code duplication, and easier to notice the `loop` inside. The downside, though, is that we check if `connection_config` is Some on every iteration, but given that it is not in a tight loop, we can trade this inefficiency for improved readability.

```suggestion
    fn input_sender_account_id(
        connection_config: Option<crate::common::ConnectionConfig>,
    ) -> color_eyre::eyre::Result<String> {
        loop {
            let account_id: String = Input::new()
                    .with_prompt("What account ID do you need to add a key?")
                    .interact_text()
                    .unwrap();
            if let Some(connection_config) = &connection_config {
                if let Some(_) = crate::common::check_account_id(
                    connection_config.clone(),
                    account_id.clone(),
                )? {
                    break Ok(account_id);
                } else {
                    println!("Account <{}> doesn't exist", account_id);
                }
            } else {
                break Ok(account_id);
            }   
        }
```